x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/kerberos/support/SimpleKdcLdapServer.java

```diff
@@ -0,0 +1,232 @@
+                init();
+                return null;
+            }
+        });
+        logger.info("SimpleKdcLdapServer started.");
+    }
+
+    @SuppressForbidden(reason = "Uses Apache Kdc which requires usage of java.io.File")
+    private void init() throws Exception {
+        // start ldap server
+        createLdapServiceAndStart();
+        // create ldap backend conf
+        createLdapBackendConf();
+        // Kdc Server
+        simpleKdc = new SimpleKdcServer(this.workDir.toFile(), new KrbConfig());
+        prepareKdcServerAndStart();
+    }
+
+    private void createLdapServiceAndStart() throws Exception {
+        InMemoryDirectoryServerConfig config = new InMemoryDirectoryServerConfig(baseDn);
+        config.setSchema(null);
+        ldapServer = new InMemoryDirectoryServer(config);
+        ldapServer.importFromLDIF(true, this.ldiff.toString());
+        ldapServer.startListening();
+        ldapPort = ldapServer.getListenPort();
+    }
+
+    private void createLdapBackendConf() throws IOException {
+        String backendConf = KdcConfigKey.KDC_IDENTITY_BACKEND.getPropertyKey()
+                + " = org.apache.kerby.kerberos.kdc.identitybackend.LdapIdentityBackend\n"
+                + "host=127.0.0.1\n" + "port=" + ldapPort + "\n" + "admin_dn=uid=admin,ou=system," + baseDn
+                + "\n" + "admin_pw=secret\n" + "base_dn=" + baseDn;
+        Files.write(this.workDir.resolve("backend.conf"), backendConf.getBytes(StandardCharsets.UTF_8));
+        assert Files.exists(this.workDir.resolve("backend.conf"));
+    }
+
+    @SuppressForbidden(reason = "Uses Apache Kdc which requires usage of java.io.File")
+    private void prepareKdcServerAndStart() throws Exception {
+        // transport
+        simpleKdc.setWorkDir(workDir.toFile());
+        simpleKdc.setKdcHost(host);
+        simpleKdc.setKdcRealm(realm);
+        if (kdcPort == 0) {
+            kdcPort = NetworkUtil.getServerPort();
+        }
+        if (transport != null) {
+            if (transport.trim().equals("TCP")) {
+                simpleKdc.setKdcTcpPort(kdcPort);
+                simpleKdc.setAllowUdp(false);
+            } else if (transport.trim().equals("UDP")) {
+                simpleKdc.setKdcUdpPort(kdcPort);
+                simpleKdc.setAllowTcp(false);
+            } else {
+                throw new IllegalArgumentException("Invalid transport: " + transport);
+            }
+        } else {
+            throw new IllegalArgumentException("Need to set transport!");
+        }
+        long minimumTicketLifeTime = simpleKdc.getKdcConfig().getMinimumTicketLifetime();
+        long maxRenewableLifeTime = simpleKdc.getKdcConfig().getMaximumRenewableLifetime();
+        simpleKdc.getKdcConfig().setLong(KdcConfigKey.MINIMUM_TICKET_LIFETIME, 86400000L);
+        simpleKdc.getKdcConfig().setLong(KdcConfigKey.MAXIMUM_RENEWABLE_LIFETIME, 604800000L);
+        logger.info("MINIMUM_TICKET_LIFETIME changed from {}  to {}", minimumTicketLifeTime,
```

Comment:
we really don't need to log this information. We can just set it here. Can you add a comment on what these values translate to in time that's human readable?